recipes/_publish.rb

```diff
@@ -0,0 +1,56 @@
+marketplace_products.each do |product|
```

Comment:
This is very delightful to look at. I'll come in and add the GCE-specific provisioner steps we need after this gets merged to master.